From 20cc3794d0ccd4bd1c696bed7f7ec2dcdfb09fe0 Mon Sep 17 00:00:00 2001
From: Charles Short <charles.short@windriver.com>
Date: Tue, 26 Oct 2021 16:59:10 +0000
Subject: [PATCH] Add wheel support

Add python3-novaclient-wheel package.

Signed-off-by: Charles Short <charles.short@windriver.com>
[Fix merge conflicts from Antelope to Caracal]
Signed-off-by: vrochalo <vinicius.rochalobo@windriver.com>
---
 debian/control | 18 ++++++++++++++++++
 debian/rules   |  2 +-
 2 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/debian/control b/debian/control
index 6e05d934..9c18db96 100644
--- a/debian/control
+++ b/debian/control
@@ -91,3 +91,21 @@ Description: client library for OpenStack Compute API - 3.x
  Compute (Nova) through the OpenStack Compute API.
  .
  This package provides the Python 3.x module.
+
+Package: python3-novaclient-wheel
+Architecture: all
+Depends:
+ python3-wheel,
+ ${misc:Depends},
+ ${python3:Depends},
+Description: client library for OpenStack Compute API - 3.x
+ OpenStack Compute, codenamed Nova, is a cloud computing fabric controller
+ designed to be modular and easy to extend and adapt. In addition to its
+ "native" OpenStack API, it also supports the Amazon EC2 API, and it supports
+ many different database backends (including SQLite, MySQL, and PostgreSQL),
+ hypervisors (KVM, Xen), and user directory systems (LDAP, SQL).
+ .
+ Python novaclient library and nova CLI tool for interacting with OpenStack
+ Compute (Nova) through the OpenStack Compute API.
+ .
+ This package contains the Python3 Wheel.
diff --git a/debian/rules b/debian/rules
index 364e3527..842222bc 100755
--- a/debian/rules
+++ b/debian/rules
@@ -14,7 +14,7 @@ override_dh_auto_build:
 	echo "Do nothing..."
 
 override_dh_auto_install:
-	pkgos-dh_auto_install --no-py2
+	pkgos-dh_auto_install --no-py2 --wheel
 	install -D -m 0644 tools/nova.bash_completion $(CURDIR)/debian/python3-novaclient/usr/share/bash-completion/completions/nova
 	rm -rf $(CURDIR)/debian/python-novaclient/usr/novaclient
 
-- 
2.34.1

